docs/proposals/distributed-tracing.md

```diff
@@ -0,0 +1,424 @@
+    - **Benefit**: Establishes gateway entry point visibility in end-to-end traces with zero configuration
+
+  * **Future Enhancement Opportunities**:
+    Component owners can add detailed spans for request parsing, model instance selection, and request/response transformations. Enhanced attributes could include token usage metrics, routing algorithms, load balancing decisions, and payload sizes.
+
+
+### Enabling Distributed Tracing
+
+The auto-instrumentation approach eliminates the need for component-specific configuration. Tracing is enabled by
+deploying an auto-instrumentation agent or operator that configures the global OpenTelemetry TraceProvider.
+
+**Triggering Auto-instrumentation:**
+This implementation supports multiple approaches for enabling tracing:
+
+1. **OpenTelemetry Operator** (Recommended for Kubernetes): Automatically injects instrumentation via annotations
+   - Documentation: [OpenTelemetry Operator](https://opentelemetry.io/docs/kubernetes/operator/)
+
+2. **Go Auto-instrumentation Agent**: Manual agent that wraps Go applications at runtime
+   - Documentation: [OpenTelemetry Go Auto-instrumentation](https://opentelemetry.io/docs/zero-code/go/)
+
+3. **Programmatic TraceProvider Setup**: Simple global tracer provider initialization for testing
+   - Documentation: [OpenTelemetry Go Manual Instrumentation](https://opentelemetry.io/docs/languages/go/getting-started/)
+
+**Auto-instrumentation Benefits:**
+- **Zero Configuration**: Components use `otel.Tracer()` calls that work with any auto-instrumentation agent
+- **Agent-Driven**: External auto-instrumentation agents provide TraceProvider configuration
+- **Platform Agnostic**: Compatible with various observability platforms and deployment methods
+- **Lightweight**: Components remain minimal with reduced tracing dependencies
+
+
+### Trace Context Propagation
+
+**Automatic Context Propagation:**
+The auto-instrumentation approach ensures trace context propagation works seamlessly across all llm-d components:
+
+* Components automatically extract incoming trace context from HTTP/gRPC headers
+* Trace context is automatically propagated to downstream service calls
+* Context is included in outgoing HTTP/gRPC headers without manual intervention
+
+This provides end-to-end trace continuity across the entire llm-d stack when an auto-instrumentation agent is active.
+
+**Performance Impact:**
+Auto-instrumentation has minimal overhead when no tracing agent is present:
+- Components use `otel.Tracer()` calls that default to no-op implementations
+- Context propagation is lightweight and stateless
+- Header extraction/injection operations are constant-time
+- No spans are created or exported without an active TraceProvider
+
+### Semantic Conventions and Attributes
+
+The implementation follows OpenTelemetry semantic conventions for GenAI operations:
+
+**Core Attributes** (implemented in auto-instrumentation):
+- `gen_ai.request.model`: Model identifier
```

Comment:
Have we tested that we can add custom attributes with auto-instrument? I can't find any document or demo to illustrate this feature.